### ISSUE TITLE
Purchases.getOfferings() has exception in ANDROID but works fine on iOS

### ISSUE THREAD

**Issue by premingiet** [RevenueCat/purchases-flutter#298]
- [x] I have updated Purchases SDK to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/RevenueCat/purchases-flutter/blob/main/CONTRIBUTING.md)
- [x] I have searched the [Community](https://community.revenuecat.com)
- [x] I have read [docs.revenuecat.com](https://docs.revenuecat.com/)
- [x] I have searched for [existing Github issues](https://github.com/RevenueCat/purchases-flutter/issues)

**Describe the bug**
I have tried both ```purchases_flutter: ^3.8.0``` as well as ```purchases_flutter: ^3.9.0``` but same issue happens.
Till yesterday everything was working well. Both on Android and iOS. But now today when i started working, i noticed, for iOS, its working well but for android when i try to use ```Purchases.getOfferings()``` i get exception ```Unhandled Exception: type 'Null' is not a subtype of type 'num' in type cast```. I am using this below method to fetch offerings. 

```dart
static Future<List<Offering>> fetchOffers() async{
    Completer<List<Offering>> _completer = Completer();
    try{
      print("================ FETCH OFFERINGS (TRY BLOCK START) ================");
      Purchases.getOfferings().then((offerings){
        print("================ DONE FETCHING OFFERINGS -> ${offerings.all} ================");
        final current = offerings.current;
        current == null ? _completer.complete([]) : _completer.complete([current]);
      });
    } on PlatformException catch(e){
      print("================ ERROR IN FETCHING OFFERINGS = $e ================");
      _completer.complete([]);
    }
    return _completer.future;
  }
```

Only on android i am getting this error and not on iOS. As you see when i use it on android, it only goes to ```TRY block``` and then when using ```Purchases.getOfferings()``` it never ends and flutter engine gives exception like this ```E/flutter (21073): <asynchronous suspension>```. I mean it doesn't even go to ```CATCH``` block.

1. Environment
   1. Output of `flutter doctor`
   
    ```dart
    Doctor summary (to see all details, run flutter doctor -v):
    [✓] Flutter (Channel beta, 2.10.0-0.1.pre, on macOS 12.1 21C52 darwin-arm, locale en-IN)
    [✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    [✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    [✓] Chrome - develop for the web
    [✓] Android Studio (version 2020.3)
    [✓] VS Code (version 1.63.2)
    [✓] Connected device (3 available)
    [✓] HTTP Host Availability
    
    • No issues found!
    ```
   2. How widespread is the issue. Percentage of devices affected.
    ```only affects android devices and not iOS```
2. [Debug logs](https://docs.revenuecat.com/docs/debugging) that reproduce the issue
```dart
D/[Purchases] - DEBUG(21073): ℹ️ Debug logging enabled
D/[Purchases] - DEBUG(21073): ℹ️ SDK Version - 4.6.0
D/[Purchases] - DEBUG(21073): 👤 Initial App User ID - null
D/[Purchases] - DEBUG(21073): 👤 Identifying App User ID: $RCAnonymousID:0fed5cd23e6d4b759b6aad08305f17c4
D/[Purchases] - DEBUG(21073): ℹ️ Deleting old synced subscriber attributes that don't belong to $RCAnonymousID:0fed5cd23e6d4b759b6aad08305f17c4
D/[Purchases] - DEBUG(21073): ℹ️ App foregrounded
D/[Purchases] - DEBUG(21073): ℹ️ PurchaserInfo cache is stale, updating from network in foreground.
D/[Purchases] - DEBUG(21073): ℹ️ Offerings cache is stale, updating from network in foreground
D/[Purchases] - DEBUG(21073): 😻 Offerings updated from network.
D/[Purchases] - DEBUG(21073): ℹ️ Skipping updating pending purchase queue since BillingClient is not connected yet.
D/[Purchases] - DEBUG(21073): ℹ️ No subscriber attributes to synchronize.
D/[Purchases] - DEBUG(21073): ℹ️ Listener set
D/[Purchases] - DEBUG(21073): ℹ️ Sending latest PurchaserInfo to listener.
D/[Purchases] - DEBUG(21073): ℹ️ Starting connection for com.android.billingclient.api.BillingClientImpl@6486156
D/[Purchases] - DEBUG(21073): ℹ️ Ending connection for com.android.billingclient.api.BillingClientImpl@80c06ef
D/[Purchases] - DEBUG(21073): ℹ️ Billing Service Setup finished for com.android.billingclient.api.BillingClientImpl@6486156
D/[Purchases] - DEBUG(21073): ℹ️ Updating pending purchase queue
D/[Purchases] - DEBUG(21073): ℹ️ API request started: GET /subscribers/%24RCAnonymousID%3A0fed5cd23e6d4b759b6aad08305f17c4
D/[Purchases] - DEBUG(21073): ℹ️ API request completed with status: GET /subscribers/%24RCAnonymousID%3A0fed5cd23e6d4b759b6aad08305f17c4 304
D/[Purchases] - DEBUG(21073): ℹ️ PurchaserInfo updated, sending to listener.
D/[Purchases] - DEBUG(21073): ℹ️ API request started: GET /subscribers/%24RCAnonymousID%3A0fed5cd23e6d4b759b6aad08305f17c4/offerings
D/[Purchases] - DEBUG(21073): ℹ️ API request completed with status: GET /subscribers/%24RCAnonymousID%3A0fed5cd23e6d4b759b6aad08305f17c4/offerings 304
D/[Purchases] - DEBUG(21073): ℹ️ Requesting products from the store with identifiers: lifetime_managed, yearly_subscription, monthly_subscription, weekly_subscription
D/[Purchases] - DEBUG(21073): ℹ️ Querying purchases
D/[Purchases] - DEBUG(21073): ℹ️ Products request finished for lifetime_managed, yearly_subscription, monthly_subscription, weekly_subscription
D/[Purchases] - DEBUG(21073): 💰 Retrieved skuDetailsList: SkuDetails: {"productId":"monthly_subscription","type":"subs","title":"Monthly Subscription Plan (SAMPLE APP)","name":"Monthly Subscription Plan","price":"₹120.00","price_amount_micros":120000000,"price_currency_code":"INR","description":"Use All features of app. You may cancel subscription anytime.","subscriptionPeriod":"P1M","skuDetailsToken":"AEuhp4KWLDH4VorTRT6Lo0FDHUKu0YyfPvI485QW8FwksLwNYAGW1Rhbp4kahcZGPlGT"}, SkuDetails: {"productId":"weekly_subscription","type":"subs","title":"Weekly Subscription Plan (SAMPLE APP)","name":"Weekly Subscription Plan","price":"₹60.00","price_amount_micros":60000000,"price_currency_code":"INR","description":"Use All features of app. You may cancel subscription anytime.","subscriptionPeriod":"P1W","skuDetailsToken":"AEuhp4KMd9iHla8uNzaR-fkKTOoMhCSvJqqNOM1NrInca95IF8I1SV8ZTxgfJAwt2G8Q"}, SkuDetails: {"productId":"yearly_subscription","type":"subs","title":"Yearly Subscription Plan (SAMPLE APP)","name":"Yearly Subscription Plan","price":"₹720.00","price_amount_micros":720000000,"price_currency_code":"INR","description":"Save 50% on yearly subscription plan. You may cancel subscription anytime.","subscriptionPeriod":"P1Y","skuDetailsToken":"AEuhp4I-O3Tse5wYgtjnRTP8m3pM7P51LOXnqqGHj-Unyvy_HjWNIE-C26cWSQq9cZm2"}
D/[Purchases] - DEBUG(21073): 💰 monthly_subscription - SkuDetails: {"productId":"monthly_subscription","type":"subs","title":"Monthly Subscription Plan (SAMPLE APP)","name":"Monthly Subscription Plan","price":"₹120.00","price_amount_micros":120000000,"price_currency_code":"INR","description":"Use All features of app. You may cancel subscription anytime.","subscriptionPeriod":"P1M","skuDetailsToken":"AEuhp4KWLDH4VorTRT6Lo0FDHUKu0YyfPvI485QW8FwksLwNYAGW1Rhbp4kahcZGPlGT"}
D/[Purchases] - DEBUG(21073): 💰 weekly_subscription - SkuDetails: {"productId":"weekly_subscription","type":"subs","title":"Weekly Subscription Plan (SAMPLE APP)","name":"Weekly Subscription Plan","price":"₹60.00","price_amount_micros":60000000,"price_currency_code":"INR","description":"Use All features of app. You may cancel subscription anytime.","subscriptionPeriod":"P1W","skuDetailsToken":"AEuhp4KMd9iHla8uNzaR-fkKTOoMhCSvJqqNOM1NrInca95IF8I1SV8ZTxgfJAwt2G8Q"}
D/[Purchases] - DEBUG(21073): 💰 yearly_subscription - SkuDetails: {"productId":"yearly_subscription","type":"subs","title":"Yearly Subscription Plan (SAMPLE APP)","name":"Yearly Subscription Plan","price":"₹720.00","price_amount_micros":720000000,"price_currency_code":"INR","description":"Save 50% on yearly subscription plan. You may cancel subscription anytime.","subscriptionPeriod":"P1Y","skuDetailsToken":"AEuhp4I-O3Tse5wYgtjnRTP8m3pM7P51LOXnqqGHj-Unyvy_HjWNIE-C26cWSQq9cZm2"}
D/[Purchases] - DEBUG(21073): ℹ️ Requesting products from the store with identifiers: lifetime_managed
D/[Purchases] - DEBUG(21073): ℹ️ Cleaning previously sent tokens
D/[Purchases] - DEBUG(21073): ℹ️ Tokens already posted: []
D/[Purchases] - DEBUG(21073): ℹ️ Saving tokens []
D/[Purchases] - DEBUG(21073): ℹ️ Tokens already posted: []
D/[Purchases] - DEBUG(21073): ℹ️ Products request finished for lifetime_managed
D/[Purchases] - DEBUG(21073): 💰 Retrieved skuDetailsList: SkuDetails: {"productId":"lifetime_managed","type":"inapp","title":"Lifetime Plan (SAMPLE APP)","name":"Lifetime Plan","price":"₹1,200.00","price_amount_micros":1200000000,"price_currency_code":"INR","description":"Pay once and use all features of app for lifetime.","skuDetailsToken":"AEuhp4J99JMFKGhMscS7u2YQPWH0NUJfQI_q1gqQgp1n6ech4ZOV3MNhE8KhD3M-snoe"}
D/[Purchases] - DEBUG(21073): 💰 lifetime_managed - SkuDetails: {"productId":"lifetime_managed","type":"inapp","title":"Lifetime Plan (SAMPLE APP)","name":"Lifetime Plan","price":"₹1,200.00","price_amount_micros":1200000000,"price_currency_code":"INR","description":"Pay once and use all features of app for lifetime.","skuDetailsToken":"AEuhp4J99JMFKGhMscS7u2YQPWH0NUJfQI_q1gqQgp1n6ech4ZOV3MNhE8KhD3M-snoe"}
I/flutter (21073): ================ FETCH OFFERINGS (TRY BLOCK START) ================
D/[Purchases] - DEBUG(21073): ℹ️ Vending Offerings from cache
D/[Purchases] - DEBUG(21073): ℹ️ Checking if cache is stale AppInBackground false
E/flutter (21073): [ERROR:flutter/lib/ui/ui_dart_state.cc(209)] Unhandled Exception: type 'Null' is not a subtype of type 'num' in type cast
E/flutter (21073): #0      _$$_IntroductoryPriceFromJson (package:purchases_flutter/models/introductory_price.g.dart:11:22)
E/flutter (21073): #1      new _$_IntroductoryPrice.fromJson (package:purchases_flutter/models/introductory_price.freezed.dart:247:7)
E/flutter (21073): #2      _$IntroductoryPriceFromJson (package:purchases_flutter/models/introductory_price.freezed.dart:17:29)
E/flutter (21073): #3      new IntroductoryPrice.fromJson (package:purchases_flutter/models/introductory_price.dart:48:7)
E/flutter (21073): #4      _$$_ProductFromJson (package:purchases_flutter/models/product_wrapper.g.dart:18:31)
E/flutter (21073): #5      new _$_Product.fromJson (package:purchases_flutter/models/product_wrapper.freezed.dart:291:7)
E/flutter (21073): #6      _$ProductFromJson (package:purchases_flutter/models/product_wrapper.freezed.dart:17:19)
E/flutter (21073): #7      new Product.fromJson (package:purchases_flutter/models/product_wrapper.dart:41:7)
E/flutter (21073): #8      _$$_PackageFromJson (package:purchases_flutter/models/package_wrapper.g.dart:13:15)
E/flutter (21073): #9      new _$_Package.fromJson (package:purchases_flutter/models/package_wrapper.freezed.dart:203:7)
E/flutter (21073): #10     _$PackageFromJson (package:purchases_flutter/models/package_wrapper.freezed.dart:17:19)
E/flutter (21073): #11     new Package.fromJson (package:purchases_flutter/models/package_wrapper.dart:76:7)
E/flutter (21073): #12     _$$_OfferingFromJson.<anonymous closure> (package:purchases_flutter/models/offering_wrapper.g.dart:13:31)
E/flutter (21073): #13     MappedListIterable.elementAt (dart:_internal/iterable.dart:413:31)
E/flutter (21073): #14     ListIterator.moveNext (dart:_internal/iterable.dart:342:26)
E/flutter (21073): #15     new _GrowableList._ofEfficientLengthIterable (dart:core-patch/growable_array.dart:206:27)
E/flutter (21073): #16     new _GrowableList.of (dart:core-patch/growable_array.dart:153:28)
E/flutter (21073): #17     new List.of (dart:core-patch/array_patch.dart:51:28)
E/flutter (21073): #18     ListIterable.toList (dart:_internal/iterable.dart:213:44)
E/flutter (21073): #19     _$$_OfferingFromJson (package:purchases_flutter/models/offering_wrapper.g.dart:14:12)
E/flutter (21073): #20     new _$_Offering.fromJson (package:purchases_flutter/models/offering_wrapper.freezed.dart:390:7)
E/flutter (21073): #21     _$OfferingFromJson (package:purchases_flutter/models/offering_wrapper.freezed.dart:17:20)
E/flutter (21073): #22     new Offering.fromJson (package:purchases_flutter/models/offering_wrapper.dart:52:7)
E/flutter (21073): #23     _$$_OfferingsFromJson.<anonymous closure> (package:purchases_flutter/models/offerings_wrapper.g.dart:12:22)
E/flutter (21073): #24     MapMixin.map (dart:collection/maps.dart:170:28)
E/flutter (21073): #25     _$$_OfferingsFromJson (package:purchases_flutter/models/offerings_wrapper.g.dart:10:28)
E/flutter (21073): #26     new _$_Offerings.fromJson (package:purchases_flutter/models/offerings_wrapper.freezed.dart:153:7)
E/flutter (21073): #27     _$OfferingsFromJson (package:purchases_flutter/models/offerings_wrapper.freezed.dart:17:21)
E/flutter (21073): #28     new Offerings.fromJson (package:purchases_flutter/models/offerings_wrapper.dart:26:7)
E/flutter (21073): #29     Purchases.getOfferings (package:purchases_flutter/purchases_flutter.dart:141:22)
E/flutter (21073): <asynchronous suspension>
```
3. Steps to reproduce, with a description of expected vs. actual behavior
    steps to reproduce:
    -> try a demo android app with everything setup in revenuecat like products, offerings, entitlements and all.
    -> Now try to use ```Purchases.getOfferings()``` to get all offerings.
    -> You will notice you are getting exception for ```Android``` only. But for ```iOS```, everything works fine
4. **Other information** (e.g. stacktraces, related issues, suggestions how to fix, links for us to have context, eg. stackoverflow, etc.)
    NONE.

**Additional context**
NONE

**Comment by signmotion** [comment]
Seems to have been fixed in #294

**Comment by aboedo** [comment]
Update here: release 3.9.1 has been released with the fix! 🎉  Thanks so much for reporting!
